packages/tools/webpack/extract-i18n-expressions-webpack-plugin.js

```diff
@@ -133,16 +126,36 @@ module.exports = class ExtractI18nExpressionsWebpackPlugin {
 
 		const translationCallExpressions = [];
 
-		this.translationsRegexps.forEach( ( regexp ) => {
-			[ ...source.matchAll( regexp ) ].forEach( ( [ callExpression ] ) => {
-				translationCallExpressions.push( callExpression );
+		[
+			...TRANSLATIONS_REGEXPS,
+			...COMMENTS_REGEXPS,
+		].forEach( ( regexp ) => {
+			[ ...source.matchAll( regexp ) ].forEach( ( res ) => {
+				translationCallExpressions.push( {
+					type: COMMENTS_REGEXPS.includes( regexp ) ? 'comment' : 'call-expression',
+					index: res.index,
+					value: res[ 0 ],
+				} );
 			} );
 		} );
 
 		return translationCallExpressions;
 	}
 
-	defaultGenerateTranslationFilename( filename ) {
-		return filename.replace( /(\.min)?\.js$/i, '.strings.js' );
+	generateTranslationFilename( basePath, filename ) {
+		return path.join(
+			basePath,
+			filename.replace( /(\.min)?\.js$/i, '.strings.js' )
+		);
+	}
+
+	generateTranslationFileContent( translationCallExpressions ) {
+		return ( translationCallExpressions || [] )
+			// Sort by the index it was founded in the file based on the regexp (and not by the order it was added to the array).
```

Comment:
```suggestion
			// Sort by the index it was found in the file based on the regexp (and not by the order it was added to the array).
```

let's talk about the index thing